### PR TITLE
Make row order optimization non-experimental

### PR DESCRIPTION
**Pull request by rschu1ze**
This PR:
- makes logging by row order optimization less verbose,
- restricts row order optimization to ordinary MergeTree tables (other MergeTree engines may break, nobody knows)
- marks row-order optimization non-experimental

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit e76fd47efbc9ac9a9a82417cab63ee34a36af1c8 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/64814/e76fd47efbc9ac9a9a82417cab63ee34a36af1c8/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>A Sync</td><td>If it fails, ask a maintainer for help</td><td>⏳ pending</td></tr>
<tbody>
</table>
<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>Checks correctness of the PR's body</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Storages/MergeTree/MergeTreeDataWriter.cpp [hunk 1]
```diff
@@ -503,7 +503,8 @@ MergeTreeDataWriter::TemporaryPart MergeTreeDataWriter::writeTempPartImpl(
             ProfileEvents::increment(ProfileEvents::MergeTreeDataWriterBlocksAlreadySorted);
     }
 
-    if (data.getSettings()->allow_experimental_optimized_row_order)
+    if (data.getSettings()->optimize_row_order
+            && data.merging_params.mode == MergeTreeData::MergingParams::Mode::Ordinary) /// Nobody knows if it the optimization messes up specialized MergeTree engines.
```
Comment (SmitaRKulkarni) [inline src/Storages/MergeTree/MergeTreeDataWriter.cpp:507]:
```suggestion
            && data.merging_params.mode == MergeTreeData::MergingParams::Mode::Ordinary) /// Nobody knows if this optimization messes up specialized MergeTree engines.
```

File: src/Storages/MergeTree/MergeTreeDataWriter.cpp [hunk 2]
```diff
@@ -730,7 +731,8 @@ MergeTreeDataWriter::TemporaryPart MergeTreeDataWriter::writeProjectionPartImpl(
             ProfileEvents::increment(ProfileEvents::MergeTreeDataProjectionWriterBlocksAlreadySorted);
     }
 
-    if (data.getSettings()->allow_experimental_optimized_row_order)
+    if (data.getSettings()->optimize_row_order
+            && data.merging_params.mode == MergeTreeData::MergingParams::Mode::Ordinary) /// Nobody knows if it the optimization messes up specialized MergeTree engines.
```
Comment (SmitaRKulkarni) [inline src/Storages/MergeTree/MergeTreeDataWriter.cpp:735]:
```suggestion
            && data.merging_params.mode == MergeTreeData::MergingParams::Mode::Ordinary) /// Nobody knows if this optimization messes up specialized MergeTree engines.
```

**Review by SmitaRKulkarni — APPROVED** [review]
Rest all LGTM

**Pull request merged**